command/agent/http_test.go

```diff
@@ -976,7 +976,7 @@ func TestHTTPServer_Limits_OK(t *testing.T) {
 		// Now assert each error is a clientside read deadline error
 		for i := 0; i < maxConns; i++ {
 			select {
-			case <-time.After(1 * time.Second):
+			case <-time.After(2 * time.Second):
```

Comment:
This timeout needs to be higher than the deadline in the goroutine above it to be correct. The test asserts the goroutine's (client's) deadline is hit before the server's, but if this is <= to both it breaks the test.